chroma_core/models/stratagem.py

```diff
@@ -0,0 +1,214 @@
+# Copyright (c) 2019 DDN. All rights reserved.
+# Use of this source code is governed by a MIT-style
+# license that can be found in the LICENSE file.
+
+import logging
+import json
+
+from django.db import models
+
+from chroma_core.lib.cache import ObjectCache
+from chroma_core.models.jobs import StatefulObject
+from chroma_core.lib.job import Step, job_log, DependOn, DependAll, DependAny
+from chroma_core.models import Job
+from chroma_core.models import StateChangeJob, StateLock
+from chroma_help.help import help_text
+from chroma_core.models import (
+    AlertStateBase,
+    AlertEvent,
+    ManagedHost,
+    ManagedMdt,
+    ManagedTarget,
+    ManagedTargetMount,
+    Volume,
+    VolumeNode,
+    StorageResourceRecord,
+)
+
+
+class StratagemConfiguration(StatefulObject):
+    id = models.IntegerField(primary_key=True, default=1, null=False)
+    interval = models.IntegerField(help_text="Interval value in seconds between each stratagem execution", null=False)
+    report_duration = models.IntegerField(
+        help_text="Interval value in seconds between stratagem report execution", null=False
+    )
+    report_duration_active = models.BooleanField(
+        default=False, help_text="Indicates if the report should execute at the given interval"
+    )
+    purge_duration = models.IntegerField(help_text="Interval value in seconds between stratagem purges", null=False)
+    purge_duration_active = models.BooleanField(
+        default=False, help_text="Indicates if the purge should execute at the given interval"
+    )
+
+    states = ["unconfigured", "configured"]
+    initial_state = "unconfigured"
+
+    class Meta:
+        app_label = "chroma_core"
+        ordering = ["id"]
+
+
+class StratagemUnconfiguredAlert(AlertStateBase):
+    default_severity = logging.ERROR
+
+    def alert_message(self):
+        return "Stratagem did not configure correctly"
+
+    class Meta:
+        app_label = "chroma_core"
+
+    def end_event(self):
+        return AlertEvent(
+            message_str="%s started" % self.alert_item,
+            alert_item=self.alert_item.primary_host,
+            alert=self,
+            severity=logging.INFO,
+        )
+
+    def affected_targets(self, affect_target):
+        affect_target(self.alert_item)
+
+
+class ConfigureStratagemTimerStep(Step):
+    def run(self, kwargs):
+        job_log.debug("Create stratagem timer step kwargs: {}".format(kwargs))
+        # Create systemd timer
+
+
+class ConfigureStratagemJob(StateChangeJob):
```

Comment:
One thing to note is that configuration needs to be bounded per filesystem. This means there must be a separate configuration / timer for each FS managed by IML.